### ISSUE TITLE
Replace usage of dependabot with go get

### ISSUE THREAD

**Issue by anuraaga** [open-telemetry/opentelemetry-collector-contrib#8541]
Go has built in support for updating `go.mod` files in a reasonable way so there doesn't seem to be that much value in dependabot. Since dependabot generates a single PR per dependency update, it spams notifications for everyone that watches this repository, a problem for maintainers and external contributors alike. It seems like it should be simple enough to remove dependabot and run `go get` just before a collector release to have an updated collector without the notification spam. Perhaps it would be a good time to look at `go work` at the same time if it helps with that and other workflows.

**Comment by mx-psi** [comment]
How would `go get` be run? A periodic Github Action?

**Comment by jpkrohling** [comment]
Just to make it clear: you are suggesting to remove dependabot only for the management of Go dependencies, right? I agree that dependabot doesn't help us in its current state. What I do like about it though is that it comes with the changelog for each change, so that we can how risky a specific change is.

I'd be interested in seeing how it works on a daily basis. Judging from my experience in other repos, there's always a dependency that would cause things to break and we wouldn't want to bump that, while all the others are desirable bumps.

**Comment by anuraaga** [comment]
> How would go get be run? A periodic Github Action?

I was thinking there would be a script that is run manually by the releaser before releasing

> Just to make it clear: you are suggesting to remove dependabot only for the management of Go dependencies, right? I agree that dependabot doesn't help us in its current state. What I do like about it though is that it comes with the changelog for each change, so that we can how risky a specific change is.

Yeah only for Go dependencies, not GHA. The note about getting the release notes conveniently is definitely a good point, but the dependabot notifications are quite a drag. If those came daily, even if it's smaller volume per day it'd still be pretty distracting at least for me...

**Comment by jpkrohling** [comment]
> I was thinking there would be a script that is run manually by the releaser before releasing

We are trying to go the other direction, leaving as few things as possible to resolve during the release time. We definitely want to sort out dependency problems before we start a release. (cc @mx-psi and @codeboten).

> , but the dependabot notifications are quite a drag

For some reason, I think Go workspaces might somehow become the answer to our problems, as a bump in a dependency that is being used by 10 modules will cause 10 PRs, when it could very well have been 1.

**Comment by mx-psi** [comment]
Agree that we want to automate as much as possible, although right now the role of Dependabot is more as a 'notify when updates are available' than a true automation, because it doesn't work well with multiple Go modules. 

This is getting off-track but: one simple way of reducing spam is to bump to aws-sdk-go-v2, since v1 releases updates at an absurdly high frequency (several per week). About ~40% of all dependabot PRs until now have been AWS SDK Go related, and the v2 SDK seems to have more different modules and less updates on any individual module. Maybe @anuraaga can tell us if switching to v2 is feasible/recommended now, and, if so, we can create a different issue to track the migration.

**Comment by jpkrohling** [comment]
I believe @codeboten has a script to open a PR with all the currently available updates. Perhaps we can have that script be executed as a scheduled action here, once or twice a week?

**Comment by anuraaga** [comment]
We recently tried aws-sdk-go-v2 in the prometheus remote write auth extension and it seems to work well and better than v1, it is appropriate to upgrade modules now I think.

That being said, I don't think this would be a solution to the dependabot spam as there are other global dependencies (e.g. testify) and there will be more components / dependencies over time, this is a problem that can only get worse, never better.

If the script can be updated to use go get to open a single PR for dependencies regularly, that would be excellent. In the meantime, I've had to go ahead and stop watching all activity in this repo since the dependabot spam is too disruptive to me. I'll still see assigned stuff so hopefully it's still OK but would rewatch if the dependabot problem could be solved.
